context/ometiff_tiler.py

```diff
@@ -18,29 +28,100 @@ def tile_ometiff(filename, channel_pages, output_directory, prefix):
+        metadata_str,
+        process_namespaces=True,
+        namespaces=namespaces
+    )['OME']['Image']['Pixels']
+    channel_metadata = image_metadata['Channel']
+    output_metadata['width'] = int(image_metadata['@SizeX'])
+    output_metadata['height'] = int(image_metadata['@SizeY'])
+    output_metadata['tile_size'] = tile_size
+    output_metadata['channels'] = {}
+    output_subdirectory = f'{prefix}.images'
+    path = os.path.join(
+        output_directory,
+        output_subdirectory
+    )
+    print(f'Writing images to {path}')
+    for channel in channel_metadata:
+        channel_info = channel['@ID'].split(':')
+        # Channel1:0
+        if(re.match('Channel([0-9]+)', channel_info[0])):
+            output_metadata['channels'][channel['@Name']] = int(channel_info[0][-1])
+        # Channel:0:1
+        else:
+            output_metadata['channels'][channel['@Name']] = int(channel_info[-1])
+    for channel in output_metadata['channels']:
+        if not os.path.exists(path):
+            os.mkdir(path)
+        image = pyvips.Image.tiffload(
+            filename, page=output_metadata['channels'][channel]
+        )
+        image.tiffsave(
+            os.path.join(path, f'{channel}.ome.tiff'),
+            strip = True,
+            tile=True,
+            tile_width=tile_size,
+            tile_height = tile_size,
+            pyramid = True,
+            compression='VIPS_FOREIGN_TIFF_COMPRESSION_DEFLATE'
+        )
+        output_metadata['channels'][channel] = os.path.join(server_url,
+            os.path.join(
+                output_subdirectory, f'{channel}.ome.tiff'
+            )
+        )
+    print(f'Writing metadata to {os.path.join(path, "tiff.json")}')
+    with open(os.path.join(path, 'tiff.json'), 'w') as outfile:
+        json.dump(output_metadata, outfile)
 
 if __name__ == '__main__':
     parser = argparse.ArgumentParser(
-        description='Create tiles with DeepZoom file.')
+        description='Create tiles with DeepZoom or TIFF pyramids.')
+    parser.add_argument(
+        '--pyramid_type',
+        help='Which kind of pyramid to use', type=PyramidType,
+        choices=list(PyramidType), required=True
+    )
     parser.add_argument(
         '--ometiff_file',
         help='OME-TIFF image to be tiled.')
     parser.add_argument(
-        '--channel_page_pairs', required=True, nargs='+',
-        help='Colon-delimited pairs of channels and page values')
+        '--channel_page_pairs', nargs="*",
+        help='Colon-delimited pairs of channels and page values',
+        default='')
     parser.add_argument(
         '--output_directory', required=True,
         help='Directory for output')
     parser.add_argument(
         '--prefix', required=True,
         help='Prefix for tile filenames')
+    parser.add_argument(
+        '--server_url', required=False, nargs="?",
+        help='Where the files will be served from'
+    )
     args = parser.parse_args()
```

Comment:
Look at https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_mutually_exclusive_group ... so we can be sure that we get a consistent set of arguments.

Reply:
Good point!

Reply:
Actually, I don't think we want them to be mutually exclusive.  Unless we add some logic to the Dockerfile, I think they should both be allowed.  Do you have a suggestion for this?